docs/sorting-algorithms/bubble-sort.mdx

```diff
@@ -0,0 +1,220 @@
+| 5 | 4 | 3 | 2 | 1 |
+
+| 4 | 5 | 3 | 2 | 1 |
+
+Next we compare value 5 with 3 and this two value must be swapped.
+| 4 | 5 | 3 | 2 | 1 |
+
+| 4 | 3 | 5 | 2 | 1 |
+
+Next we compare value 5 with 2 and this also must be swapped.
+| 4 | 3 | 5 | 2 | 1 |
+
+| 4 | 3 | 2 | 5 | 1 |
+
+Then we move to the next two value which is 5 and 1, we swapped.
+| 4 | 3 | 2 | 5 | 1 |
+
+| 4 | 3 | 2 | 1 | 5 |
+
+Now 5 is bubbled.
+
+Notice that after each iteration, at least one value moves at the end.
+
+Likewise we are doing rest of passes to sort the array completely.
+Pass 2.
+
+| 4 | 3 | 2 | 1 |
+
+| 3 | 4 | 2 | 1 |
+
+| 3 | 2 | 4 | 1 |
+
+| 3 | 2 | 1 | 4 |
+
+Now 4 is bubbled.
+
+Pass 3.
+
+| 3 | 2 | 1 |
+
+| 2 | 3 | 1 |
+
+| 2 | 1 | 3 |
+
+Now 3 is bubbled.
+
+Pass 4.
+
+| 2 | 1 |
+
+| 1 | 2 |
+
+Now 2 is bubbled.
+
+So now our array now looks like::  1 2 3 4 5
+```
+
+## Algorithm
+
+- Read total number of elements say `n`.
+- Store the elements in an array say `arr`.
+- Initially `i = 0`.
+- Compare adjacent elements.
+- Repeat step `4` for all `n` elements.
+- Increment `i` by `1` and repeat steps `4,5` for `i < n`.
+
+## Code
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include <iostream>
+using namespace std;
+
+//Function for swapping two elements
+void swap(int *a, int *b)
+{
+    int temp = *a;
+    *a = *b;
+    *b = temp;
+}
+
+//Function for sorting the array
+void bubbleSort(int arr[], int n)
+{
+    //outer loop for number of Passes
+    for (int i = 0; i < n - 1; i++)
+    {
+        //inner loop for number of comparisions
+        for (int j = 0; j < n - 1 - i; j++)
+        {
+            //swapping adjacent elements if they are in wrong order
+            if (arr[j] > arr[j + 1])
+                swap(&arr[j], &arr[j + 1]);
+        }
+    }
+}
+
+int main()
+{
+    int n;
+    cin >> n;
+    int arr[n];
+
+    for (int i = 0; i < n; i++)
+        cin >> arr[i];
+
+    bubbleSort(arr, n);
+
+    for (int i = 0; i < n; i++)
+        cout << arr[i] << ' ';
+
+    return 0;
+}
+```
+
+</TabItem>
+<TabItem value="py">
+
+```py
+# Function for sotring the array
+def bubbleSort(arr):
+
+    # outer loop for number of passes
+    for i in range(len(arr)):
+
+        # inner loop for number of comparisons
+        for j in range(len(arr) - 1 - i):
+            # swapping elements if they are in wrong order
+            if arr[j] > arr[j+1]:
+                temp = arr[j]
+                arr[j] = arr[j+1]
+                arr[j+1] = temp
+
+
+arr = []
+size = int(input("Enter the size of list :: "))
+
+for i in range(0, size):
+    element = int(input("Enter element : "))
+
+    arr.append(element)
+
+print("\nyou entered   :: ", arr)
+
+# sorting list
+bubbleSort(arr)
+
+print("After sorting :: ", arr)
```

Comment:
These input statements are not present in C++ code. Make I/O similar across both codes.

Reply:
okay